### PR TITLE
Add additional data in voice broadcast events 

### PR DESCRIPTION
**Pull request by Florian14**
## Type of change

- [x] WIP Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Add `io.element.voice_broadcast_chunk` in voice message content
Add `device_id` in `io.element.voice_broadcast_info` state event
Add entries in matrix-sdk to add additional content fields to the events

## Motivation and context

Continue #7127 and follow the "specification" defined in https://github.com/vector-im/element-meta/discussions/632 

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/voicebroadcast/usecase/StartVoiceBroadcastUseCase.kt [hunk 1]
```diff
@@ -98,7 +100,10 @@ class StartVoiceBroadcastUseCase @Inject constructor(
                 attachment = audioType.toContentAttachmentData(isVoiceMessage = true),
                 compressBeforeSending = false,
                 roomIds = emptySet(),
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/voicebroadcast/usecase/StartVoiceBroadcastUseCase.kt:102]: Is it ok to have roomIds empty here?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/send/SendService.kt [hunk 2]
```diff
@@ -45,18 +45,30 @@ interface SendService {
...
      * @param autoMarkdown If true, the SDK will generate a formatted HTML message from the body text if markdown syntax is present
+     * @param additionalContent additional content to put in the event content
      * @return a [Cancelable]
      */
-    fun sendTextMessage(text: CharSequence, msgType: String = MessageType.MSGTYPE_TEXT, autoMarkdown: Boolean = false): Cancelable
+    fun sendTextMessage(
+            text: CharSequence,
+            msgType: String = MessageType.MSGTYPE_TEXT,
+            autoMarkdown: Boolean = false,
+            additionalContent: Content? = null,
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/send/SendService.kt:55]: It's a bit weird to have this available in all methods but I guess its acceptable.

**Review by ganfra — COMMENTED** [review]
Some small remarks

File: vector/src/main/java/im/vector/app/features/voicebroadcast/usecase/StartVoiceBroadcastUseCase.kt [hunk 3]
```diff
@@ -95,7 +95,6 @@ class StartVoiceBroadcastUseCase @Inject constructor(
                 "Voice Broadcast Part ($sequence).${voiceMessageFile.extension}"
         )
         val audioType = outputFileUri.toMultiPickerAudioType(context) ?: return
-        // TODO put sequence in event content
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/voicebroadcast/usecase/StartVoiceBroadcastUseCase.kt:96]: +1

**Review by ganfra — APPROVED** [review]
Ok for me then

**Comment by Florian14** [comment]
Just did two small fixes before merging

**Comment by ganfra** [comment]
Can you look at the failing tests?

**Pull request merged**